samcli/lib/telemetry/event.py

```diff
@@ -145,6 +154,26 @@ def clear_trackers():
         with EventTracker._event_lock:
             EventTracker._events = []
 
+    @staticmethod
+    def send_events():
+        """Sends the current list of events via Telemetry."""
+        from samcli.lib.telemetry.metric import Metric  # pylint: disable=cyclic-import
+
+        if not EventTracker.get_tracked_events():  # Don't do anything if there are no events to send
+            return
+
+        telemetry = Telemetry()
+
+        try:
+            metric = Metric("events")
+            msa = {}
+            msa["events"] = [e.to_json() for e in EventTracker.get_tracked_events()]
+            metric.add_data("metricSpecificAttributes", msa)
+            telemetry.emit(metric)
+            EventTracker.clear_trackers()
+        except RuntimeError:
+            LOG.debug("Unable to find Click Context for getting session_id.")
```

Comment:
discussed with Leo in a huddle and we decided we should probably do all this while holding a lock in order to avoid race conditions (e.g. sending duplicate  events, etc.), as this function is called without holding a lock